lib/component-data/actions.js

```diff
@@ -148,12 +148,17 @@ function clientSave(uri, data, oldData, store, {eventID, snapshot, paths}) { //
     .then((savedData) => publish(uri, savedData, eventID, snapshot, store))
     .then((savedData) => modelRender(uri, savedData))
     .then((renderableData) => {
-      const fields = _.reduce(renderableData, (result, val, key) => _.isEqual(val, oldData[key]) ? result : result.concat(key), []);
+      if (oldData) {
+        const fields = _.reduce(renderableData, (result, val, key) => _.isEqual(val, oldData[key]) ? result : result.concat(key), []);
 
-      store.commit(UPDATE_COMPONENT, {uri, data: renderableData, fields});
-      return store.dispatch('render', { uri, data: renderableData, snapshot, paths });
+        store.commit(UPDATE_COMPONENT, {uri, data: renderableData, fields});
+        return store.dispatch('render', { uri, data: renderableData, snapshot, paths });
```

Comment:
This process is repeated in the `if` and in the `else` clause, you can avoid duplication having it outside the `if...else...`

Reply:
👍